src/DependencyInjection/Providers/AppIdProviderConfigurator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace KnpU\OAuth2ClientBundle\DependencyInjection\Providers;
+
+use Symfony\Component\Config\Definition\Builder\NodeBuilder;
+use Jampire\OAuth2\Client\Provider\AppIdProvider;
+use KnpU\OAuth2ClientBundle\Client\Provider\AppIdClient;
+
+/**
+ * Class AppIdProviderConfigurator
+ *
+ * @author Dzianis Kotau <jampire.blr@gmail.com>
+ * @package KnpU\OAuth2ClientBundle\DependencyInjection\Providers
+ */
+class AppIdProviderConfigurator implements ProviderConfiguratorInterface
+{
+
+    public function buildConfiguration(NodeBuilder $node)
+    {
+        $node
+            ->scalarNode('base_auth_uri')
+                ->isRequired()
+                ->info('IBM App ID base URL. More detail: https://cloud.ibm.com/docs/services/appid?topic=appid-getting-started')
+                ->example('base_auth_uri: https://us-south.appid.cloud.ibm.com/oauth/v4')
+            ->end()
+            ->scalarNode('tenant_id')
+                ->isRequired()
+                ->info('IBM App ID service tenant ID. More detail: https://cloud.ibm.com/docs/services/appid?topic=appid-getting-started')
+                ->example('tenant_id: 1234-5678-abcd-efgh')
+            ->end()
+            ->scalarNode('idp')
+                ->defaultValue('saml')
+                ->info('Identity Provider code. More detail: https://cloud.ibm.com/docs/services/appid?topic=appid-getting-started')
+                ->example('idp: saml')
+            ->end()
+        ;
+    }
+
+    public function getProviderClass(array $configuration)
+    {
+        return AppIdProvider::class;
+    }
+
+    public function getClientClass(array $config)
+    {
+        return AppIdClient::class;
```

Comment:
This should be a "string" value, otherwise, we will have "class not found" error, but we don't want to enforce users to have `jampire/oauth2-appid` even if they don't use it. See as example: https://github.com/knpuniversity/oauth2-client-bundle/blob/9997fbec0ad46d6420e0cfd95486258c5623343e/src/DependencyInjection/Providers/FacebookProviderConfigurator.php